Extension/lib/utils/common.js

```diff
@@ -337,3 +337,102 @@ StopWatch.prototype = {
         console.log(this.name + "[elapsed: " + elapsed + " ms]");
     }
 };
+
+/**
+ * Utility class for saving and retrieving some details by key;
+ * It's bounded with some capacity.
+ * Details are stored in some ring buffer. For each key corresponding details are retrieved in LIFO order.
+ */
+var DetailsStorage = exports.DetailsStorage = (function () {
+
+    if (typeof Map === 'undefined') {
+        return;
+    }
+
+    /**
+     * detailsKeyToIndex: Map (key => indexes)
+     * indexes = Array of [index];
+     * index = position of details in detailsRingBuffer
+     */
+    /* global Map */
+    var detailsKeyToIndex = new Map();
+    var detailsWritePointer = 0; // Current write position
+
+    /**
+     * detailsRingBuffer: Array [0:details][1:details]...[255:details]
+     */
+    var detailsRingBuffer = new Array(256);
+
+    var i = detailsRingBuffer.length;
+    while (i--) {
+        detailsRingBuffer[i] = {processedKey: null}; // 'if not null' means this details hasn't been processed yet.
+    }
+
+    /**
+     * Creates and store new details
+     * 1. Associates details with next index from detailsRingBuffer.
+     * 2. If index has been already in use and details hasn't been processed yet, then removes it from indexes array in detailsKeyToIndex
+     * 3. Push this index to indexes array in detailsKeyToIndex at first position
+     * @param key Key
+     */
+    var create = function (key) {

```

Comment:
Ok, I'll rename it. But ```value``` is redundant parameter. The main idea is reuse preallocated objects and avoid unnecessary allocations.